stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -46,81 +53,102 @@ namespace stan {
                     (A.rows() * (A.rows() + 1) / 2)),
           variRefL_(ChainableStack::memalloc_.alloc_array<vari*>
                     (A.rows() * (A.rows() + 1) / 2)) {
-        size_t accum = 0;
-        size_t accum_i = accum;
-        for (size_type j = 0; j < M_; ++j) {
-          for (size_type i = j; i < M_; ++i) {
-            accum_i += i;
-            size_t pos = j + accum_i;
-            variRefA_[pos] = A.coeffRef(i, j).vi_;
-            variRefL_[pos] = new vari(L_A.coeffRef(i, j), false);
+            size_t pos = 0;
+            block_size_ = M_/8;
+            block_size_ = (block_size_/16)*16;
+            block_size_ = (std::min)((std::max)(block_size_, 8), 128);
+            for (size_type j = 0; j < M_; ++j) {
+              for (size_type i = j; i < M_; ++i) {
+                variRefA_[pos] = A.coeffRef(i, j).vi_;
+                variRefL_[pos] = new vari(L_A.coeffRef(i, j), false);
+                ++pos;
+              }
+            }
           }
-          accum += j;
-          accum_i = accum;
-        }
+
+      inline void symbolic_rev(Block_& L,
+                               Block_& Lbar,
+                               int size) {
+        using Eigen::Lower;
+        using Eigen::Upper;
+        using Eigen::StrictlyUpper;
+        L.transposeInPlace();
+        Lbar = (L * Lbar.triangularView<Lower>()).eval();
+        Lbar.triangularView<StrictlyUpper>()
+          = Lbar.adjoint().triangularView<StrictlyUpper>();
+        L.triangularView<Upper>().solveInPlace(Lbar);
+        L.triangularView<Upper>().solveInPlace(Lbar.transpose());
```

Comment:
Not a linear algebra expert, but solving this in place twice seems odd ot me.  What's this computing?